R/bundle.R

```diff
@@ -923,20 +923,24 @@ explodeFiles <- function(dir, files) {
 
 # Run "quarto inspect" on the target and returns its output as a parsed object.
 quartoInspect <- function(appDir = NULL, appPrimaryDoc = NULL, quarto = NULL) {
-  if (!is.null(quarto)) {
-    inspect <- NULL
-    # If "quarto inspect appDir" fails, we will try "quarto inspect
-    # appPrimaryDoc", so that we can support single files as well as projects.
-    primaryDocPath <- file.path(appDir, appPrimaryDoc) # prior art: appHasParameters()
-    for (path in c(appDir, primaryDocPath)) {
-      args <- c("inspect", path.expand(path))
-      inspect <- suppressWarnings(system2(quarto, args, stdout = TRUE, stderr = FALSE))
-      if (is.null(attr(inspect, "status"))) {
-        return(jsonlite::fromJSON(inspect))
-      }
-    }
+  if (is.null(quarto)) {
+    return(NULL)
+  }
+  inspect <- NULL
+  # If "quarto inspect appDir" fails, we will try "quarto inspect
+  # appPrimaryDoc", so that we can support single files as well as projects.
+  primaryDocPath <- file.path(appDir, appPrimaryDoc) # prior art: appHasParameters()
+  for (path in c(appDir, primaryDocPath)) {
+    args <- c("inspect", path.expand(path))
+    tryCatch(
+      {
+        inspectOutput <- suppressWarnings(system2(quarto, args, stdout = TRUE, stderr = TRUE))
+        inspect <- jsonlite::fromJSON(inspectOutput)
+        return(inspect)
+      },
+      error = function(e) e
```

Comment:
What value is returned by this function when both inspect attempts err?

Reply:
It should be `NULL`.

The function was originally written with the pattern of (pseudocode):

```
inspect <- NULL
inspect <- quarto inspect
return(inspect)
```

Checking the `appDir` and then the `appPrimaryDoc` in sequence, I added `return(inspect)` inside the loop so that we didn't call `quarto inspect` when we didn't need to, because it's expensive. I think I should change the main portion of the function to either (1) continue to return `inspect` within the loop, and return `NULL` at the end, or (2) assign `inspect <- NULL` at the start, break out of the loop if `inspect` is found, return `inspect` at the end.

Which do you think would be better? Or do you think there's a clearer approach I haven't considered?

Reply:
From looking around the codebase, I think (2).